internal/webhook/kubernetes/stage/webhook.go

```diff
@@ -234,3 +251,54 @@ func (w *webhook) validateRequestedFreight(
 	}
 	return nil
 }
+
+// validatePromotionStepTaskRefs validates that PromotionTemplate steps that
+// reference a task do not have an 'if' condition or a config field set.
+//
+// Note that this validation is specific to Stages, as other resources (e.g.
+// Promotions and PromotionTasks) do not allow steps that reference a task
+// at all.
+func (w *webhook) validatePromotionStepTaskRefs(
+	f *field.Path,
+	steps []kargoapi.PromotionStep,
+) field.ErrorList {
+	errs := field.ErrorList{}
+	for i, step := range steps {
+		if step.Task == nil {
+			continue
+		}
+
+		// TODO(hidde): 'if' is a reserved CEL keyword, and because of this,
+		// validation rules at CRD level are not allowed on Kubernetes versions
+		// below 1.31.x. Once the minimum supported Kubernetes version is 1.31.x,
+		// this can be changed to a validation rule at CRD level.
+		//
+		// xref: https://github.com/akuity/kargo/pull/4732
+		// xref: https://github.com/kubernetes/kubernetes/pull/126977
+		if step.If != "" {
+			errs = append(
+				errs,
+				field.Forbidden(
+					f.Index(i).Child("if"),
+					"PromotionTemplate step referencing a task cannot have an 'if' condition",
+				),
```

Comment:
This uses "forbidden" because "invalid" would reflect the value, which appeared to be impractical for the config validation below. This still produces a nice error, however:

```
kubectl apply -f kargo.yaml
project.kargo.akuity.io/kargo-demo-06 created
projectconfig.kargo.akuity.io/kargo-demo-06 created
warehouse.kargo.akuity.io/kargo-demo created
promotiontask.kargo.akuity.io/promo-process created
stage.kargo.akuity.io/test created
stage.kargo.akuity.io/prod created
analysistemplate.argoproj.io/kargo-demo created
The Stage "uat" is invalid: spec.promotionTemplate.spec.steps[0].if: Forbidden: PromotionTemplate step referencing a task cannot have an 'if' condition
```